### ISSUE TITLE
Is there plans to add the Timezones America/Bahia and America/Manaus (From Brazil) to ActiveSupport::TimeZone.all?

### ISSUE THREAD

**Issue by pedrofurtado** [rails/rails#35341]
Today, in our Rails application, we need to deal with 3 timezones in Brazil:

* America/Sao_Paulo or America/Brasilia (are the same and it is included by default in Rails)
* America/Bahia (not included yet)
* America/Manaus (not included yet)

Thanks! :+1:

**Comment by marceloperini** [comment]
@rafaelfranca what do you think? I can work with this issue if it is relevant

**Comment by byellokore** [comment]
I think that it's not a issue. These two time zones  are already implemented inside TZInfo gem. I submitted the PR yesterday but I'm not sure if it will be accepted.  [https://github.com/rails/rails/pull/35347](url)

**Comment by marceloperini** [comment]
> I think that it's not a issue. These two time zones are already implemented inside TZInfo gem. I submitted the PR yesterday but I'm not sure if it will be accepted. [https://github.com/rails/rails/pull/35347](url)

Your link is broken, try this link [https://github.com/rails/rails/pull/35347](https://github.com/rails/rails/pull/35347)

**Comment by rafaelfranca** [comment]
Duplicate of #22440.

**Comment by rafaelfranca** [comment]
👋 I closed this as duplicated because we don't want to add new timezone to that list of mapping. Why do you need to add timezone to `ActiveSupport::TimeZone.all`?

It is because you want to build a select with the options for those timezones?

**Comment by pedrofurtado** [comment]
@rafaelfranca Yes, I want to build a select with the options for those timezones and also use the timezone to make calculations with dates, datetimes and times.

**Comment by rafaelfranca** [comment]
`ActiveSupport::TimeZone.all` is not used for the calculations. You can do it without adding to that list. About the select I think you can build it yourself easily based on `ActiveSupport::TimeZone.all` + all the other zones you want.

**Comment by pedrofurtado** [comment]
@rafaelfranca 
It is worth not add this two timezones despite of a small increase of maintance overhead, considering the facility provided for all Brazilian applications and developers? Yes, we can make some workaround in our application, that's true, and that is not difficult. But if the framework can provide it by default, and this feature is a not a nightmare to maintain, I think this could be a good idea.

Anyway, thanks for your feedback and your work in Rails! :+1: :handshake:

**Comment by pedrofurtado** [comment]
Anyway, for future readers of this issue, the hack that I made was the following:

* Create a file called, for example, `lib/core_extensions/my_timezones.rb`, with the following code:

```ruby
# frozen_string_literal: true

ActiveSupport::TimeZone::MAPPING['Bahia'] = 'America/Bahia'
ActiveSupport::TimeZone::MAPPING['Manaus'] = 'America/Manaus'
```

Then, create a file called `config/initializers/core_extensions.rb`, with the following code:

```ruby
# frozen_string_literal: true

Dir[Rails.root.join('lib/core_extensions/**/*.rb')].each { |f| require f }
```

Then, that's right!

**Comment by rafaelfranca** [comment]
I do agree this needs to be easier but I don't think we should add to the current mapping. As @pixeltrix mentioned in https://github.com/rails/rails/issues/10276#issuecomment-33014344 I think we should make the timezone select manageable using i18n so people can add new timezones and also translate their selects.

**Comment by pedrofurtado** [comment]
> I do agree this needs to be easier but I don't think we should add to the current mapping. As @pixeltrix mentioned in [#10276 (comment)](https://github.com/rails/rails/issues/10276#issuecomment-33014344) I think we should make the timezone select manageable using i18n so people can add new timezones and also translate their selects.

@rafaelfranca That is a good idea :handshake:

**Comment by pixeltrix** [comment]
This is trivial todo today - just create a locale file somewhat like this:

``` yaml
en:
  time_zones:
    menu:
      "US & Canada":
        - ["Hawaii (-10:00)", "Pacific/Honolulu"]
        - ["Alaska (-09:00)", "America/Juneau"]
        - ["Pacific Time (-08:00)", "America/Los_Angeles"]
        - ["Mountain Time (-07:00)", "America/Denver"]
        - ["Arizona (-07:00)", "America/Phoenix"]
        - ["Central Time (-06:00)", "America/Chicago"]
        - ["Eastern Time (-05:00)", "America/New_York"]
        - ["Indiana (-05:00)", "America/Indiana/Indianapolis"]
      "Mexico":
        - ["Tijuana (-08:00)", "America/Tijuana"]
        - ["Chihuahua (-07:00)", "America/Chihuahua"]
        - ["Mazatlan (-07:00)", "America/Mazatlan"]
        - ["Mexico City (-06:00)", "America/Mexico_City"]
        - ["Monterrey (-06:00)", "America/Monterrey"]
      "Europe":
        - ["London (+00:00)", "Europe/London"]
        - ["Madrid (+01:00)", "Europe/Madrid"]
        - ["Paris (+01:00)", "Europe/Paris"]
        - ["Berlin (+01:00)", "Europe/Berlin"]
        - ["Rome (+01:00)", "Europe/Rome"]
      "Asia":
        - ["Hong Kong (+08:00)", "Asia/Hong_Kong"]
        - ["Seoul (+09:00)", "Asia/Seoul"]
        - ["Tokyo (+09:00)", "Asia/Tokyo"]
```

and then use it in a form like this:

``` erb
<%= form.grouped_select :time_zone, t(:menu, scope: :time_zones) %>
```

This gives a much more pleasing menu than the built-in `time_zone_select`:

![image](https://user-images.githubusercontent.com/6321/53446586-3fbe5400-3a0b-11e9-94e1-763a488df58b.png)

Also, by storing the time zone using the zone identifier string like `Europe/London` rather than the `ActiveSupport::TimeZone::MAPPING` name (e.g. `London`) it bypasses the issue with any particular time zone not being in the predefined list.

**Comment by pedrofurtado** [comment]
@pixeltrix Thanks for feedback! Is this available in Rails 5.2.2? Or only in Rails 6.x?

**Comment by pixeltrix** [comment]
@pedrofurtado sorry, yes - completely forgot `grouped_select` is a custom form builder method I use - it's just `select` - sorry for the copy/paste confusion.
